### PR TITLE
Makes pulling money out of IDs more consistent, adds actually useful feedback to pulling money out of cards

### PR DESCRIPTION
**Pull request by MrDoomBringer**
## About The Pull Request

Makes money stuff more consistant
also adds some messages when you try to withdraw showing how much money you have, and if failed, how big the difference was between the withdrawal attempt amount and the bank accounts balance

## Why It's Good For The Game

consistency = good

## Changelog
:cl:
tweak: You no longer have to hold you ID in your active hand to pull out dosh
tweak: Cards now tell you more about your account when withdrawing cash
/:cl:

### CODE REVIEW

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -174,10 +174,9 @@
 	if(!isliving(user))
 		return
 	if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
-		return
-	if(user.get_active_held_item() != src)
 		to_chat(user, "<span class='warning'>You must hold the ID in your hand to do this.</span>")
```
Comment (Tlaltecuhtli) [inline code/game/objects/items/cards_ids.dm]: You must hold the ID in your hand to do this.

Comment (MrDoomBringer) [reply code/game/objects/items/cards_ids.dm]: yeah i was just about to fix this re

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -174,10 +174,9 @@
 	if(!isliving(user))
 		return
 	if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
```
Comment (ShizCalev) [inline code/game/objects/items/cards_ids.dm]: canUseTopic already handles this message.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/cards_ids.dm [hunk 3]
```diff
@@ -210,7 +209,8 @@
 		to_chat(user, "<span class='notice'>You withdraw [amount_to_remove] credits into a holochip.</span>")
 		return
 	else
-		to_chat(user, "<span class='warning'>The linked account doesn't have the funds for that.</span>")
+		var/difference = amount_to_remove - registered_account.account_balance
+		to_chat(user, "<span class='warning'>The linked account needs [difference] more credit(s) to perform that withdrawal.</span>")
```
Comment (ShizCalev) [inline code/game/objects/items/cards_ids.dm]: credit\s

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/cards_ids.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/game/objects/items/cards_ids.dm]: fukj

File: code/game/objects/items/cards_ids.dm (hunk 3 above)
Comment (MrDoomBringer) [reply code/game/objects/items/cards_ids.dm]: really? how the fuck

File: code/game/objects/items/cards_ids.dm [hunk 4]
```diff
@@ -210,7 +209,8 @@
 		to_chat(user, "<span class='notice'>You withdraw [amount_to_remove] credits into a holochip.</span>")
 		return
 	else
-		to_chat(user, "<span class='warning'>The linked account doesn't have the funds for that.</span>")
+		var/difference = amount_to_remove - registered_account.account_balance
+		to_chat(user, "<span class='warning'>The linked account needs [difference] more credit\s to perform that withdrawal.</span>")
```
Comment (MacHac) [inline code/game/objects/items/cards_ids.dm]: Consider using `registered_account.bank_card_talk(msg, TRUE)` here to make sure economy-related feedback stays consistent.

Comment (MrDoomBringer) [reply code/game/objects/items/cards_ids.dm]: IDK, i feel like there's no benefit to not receiving this info if you're deaf. Also, this is consistent in terms of feedback via `to_chats` for things regarding user input

Comment (MacHac) [reply code/game/objects/items/cards_ids.dm]: I was envisioning more of an ATM vibe (which the bank_card_talk format would fit better), but you could make a decent argument for either way.  It's probably fine.

Comment (ShizCalev) [reply code/game/objects/items/cards_ids.dm]: yeah true here would be good

Comment (MrDoomBringer) [reply code/game/objects/items/cards_ids.dm]: well alright

**Pull request merged**